### ISSUE TITLE
can't fetch host key

### ISSUE THREAD

**Issue by MartinL0815** [hbons/SparkleShare#1098]
I'm completely new to github and to ssh, but the ssh URL github provides for read and write access to my newly created project is slightly different to the one "compiled" by SparkleShare:

```
git@github.com:MartinLemburg/WebPageProject.git
```

The "compiled" one:

```
git@github.com/MartinLemburg/WebPageProject.git
```

Since this delimeter between server and the path is not part of the plugin XML configuration it perhaps doesn't cause the problem SparkleShare has, right?

**Comment by teelittle** [comment]
Hi there, 
our organisation is using SparkleShare with Planio (hosted redmine). 
We had the same issue (`error: could not fetch host key`) with a new installation of SparkleShare in Windows, and I would like to share my findings (and solution ;-)). 

- after installation of SparkleShare (either 1.4 or 1.5 for Windows) we tried to add a project (RMB on taskbar icon `> Sparkleshare > add hosted project...`) 
- SparkleShare fails with `error: could not fetch host key`
- detailed error documentation is found in `%appdata%\SparkleShare\logs\debug_log_YYYY_MM_DD.N.txt` 
  - you should know that SparkleShare has a well selected toolbox (git, ssh, ping, vi, ...) installed in the `C:\Program Files (x86)\SparkleShare\msysgit\bin\` subdirectory
  - you should also know that the SSH suite is used by SparkleShare to establish the initial connection with the remote repository server (therefore, it installes certain SSH tools in its toolbox directory) 
- the lastest log reveals that ssh-keyscan failed saying `<hostname>: exception!`
  `ssh-keyscan` throws his error message when no connection can be established with the server 
- when adding a new project, SparkleShare employs `ssh-keyscan.exe` to fetch the repository server's public key needed to make the SSH connection 
  - please note: SSH addresses port 22 on the repository server

Now, when there is absolutely no connection possible with the repository server (and you can exclude typos in the server's and project's URL/address), most likely port 22 is blocked on your side. 
This may be caused by your firewall, a virus scanner (really??), or your (WLAN) router. 

In our case, the machine where SparkleShare was to be installed had been taked to the private home of the admin to carry out installation and initial configuration. The admin had connected the laptop to his home's public WLAN (instead of his private WLAN). 
Our finding is that many WLAN routers offering a private and a public WLAN do not allow port 22 for the public WLAN. 

After connecting said laptop to the private WLAN (without any access restrictions), we hit Sparkleshare's "Try again" button, and adding the git project worked like a charm! 

To test if connection to your repository server is blocked or available, open a command prompt and enter: 
`C:\Program Files (x86)\SparkleShare\msysgit\bin\ssh-keyscan -t rsa1,rsa,dsa -p 22 [IP or DNS name of repository server]`
This will fetch any available public key from that server (the version of ssh-keyscan included with SparkleShare 1.5 for windows seems to support no more key algorithms). 
- if you get `[server address]: exception!`, no connection could be made 
- a correct output should look like this (example: plan.io): 

`C:\Program Files (x86)\SparkleShare\msysgit\bin>ssh-keyscan -t rsa1,rsa,dsa -p 22 plan.io`
`# plan.io SSH-2.0-OpenSSH_6.7p1 Debian-5+deb8u5`
`no hostkey alg`
`# plan.io SSH-2.0-OpenSSH_6.7p1 Debian-5+deb8u5`
`plan.io ssh-rsa AAAAB3NzaC1yc2EAAAABIwAAAQEAx5Fomonm3dlFAleTKSoMiAQ0MjtsA/QrvD0i`
`yZ2xnP5gdu4ga1vQNFTz0aS7jnk9GPazg6zuJ/DPJ9c0omXOZ4i9vGX0h5OrsHlmUStpj2xUe9rt/ebs`
`9ag20hV6+F8argBz6PHBLA1cyULfpGsmgrZFDslngot58HhErG9S2pQo88OiYxx7l1eCkjDnzFE4u0Sy`
`HA/mdJcuGIEfqUNQL+4BWQuXARAUKkrRmM+r+HCFGWlYMV/oHQdbKMo/nXVH2I1ra4TZgCdx1EBLhJIB`
`ebrwF4dFiSg2z0sH2IItg9k7bprUE2dWcGpZl3XR+vyCfUJqUKqokfW9HcY7rhpOfQ==`

**Comment by hbons** [comment]
@teelittle thanks for documenting your findings! :)

**Comment by DemonRem** [comment]
wonder why 2.0 version not posted on the website for windows -.-"

**Comment by kjyv** [comment]
I noticed I also receive no output of ssh-keyscan unless I run it with -4 (on macOS 12.3). Alternatively, disabling IPv6 for the current network did the trick as well. Not sure if the server needs additional configuration to make this work with default settings.
